server/src/com/cloud/api/ApiResponseHelper.java

```diff
@@ -526,16 +529,18 @@ public static DataStoreRole getDataStoreRole(Snapshot snapshot, SnapshotDataStor
         }
 
         long storagePoolId = snapshotStore.getDataStoreId();
-        DataStore dataStore = dataStoreMgr.getDataStore(storagePoolId, DataStoreRole.Primary);
+        if (! snapshotStore.getState().equals(ObjectInDataStoreStateMachine.State.Destroyed)){
```

Comment:
@nvazquez this can cause NPE, instead use `ObjectInDataStoreStateMachine.State.Destroyed.equals(...)` or an explicit != null.